// es-abstract/es2020.js
'use strict';

/* eslint global-require: 0 */
// https://www.ecma-international.org/ecma-262/10.0/#sec-abstract-operations
var ES2020 = {
	'Abstract Equality Comparison': require('es-abstract/2020/AbstractEqualityComparison'),
	'Abstract Relational Comparison': require('es-abstract/2020/AbstractRelationalComparison'),
	'Strict Equality Comparison': require('es-abstract/2020/StrictEqualityComparison'),
	abs: require('es-abstract/2020/abs'),
	AddEntriesFromIterable: require('es-abstract/2020/AddEntriesFromIterable'),
	AdvanceStringIndex: require('es-abstract/2020/AdvanceStringIndex'),
	ArrayCreate: require('es-abstract/2020/ArrayCreate'),
	ArraySetLength: require('es-abstract/2020/ArraySetLength'),
	ArraySpeciesCreate: require('es-abstract/2020/ArraySpeciesCreate'),
	BigIntBitwiseOp: require('es-abstract/2020/BigIntBitwiseOp'),
	BinaryAnd: require('es-abstract/2020/BinaryAnd'),
	BinaryOr: require('es-abstract/2020/BinaryOr'),
	BinaryXor: require('es-abstract/2020/BinaryXor'),
	Call: require('es-abstract/2020/Call'),
	CanonicalNumericIndexString: require('es-abstract/2020/CanonicalNumericIndexString'),
	CodePointAt: require('es-abstract/2020/CodePointAt'),
	CompletePropertyDescriptor: require('es-abstract/2020/CompletePropertyDescriptor'),
	CopyDataProperties: require('es-abstract/2020/CopyDataProperties'),
	CreateDataProperty: require('es-abstract/2020/CreateDataProperty'),
	CreateDataPropertyOrThrow: require('es-abstract/2020/CreateDataPropertyOrThrow'),
	CreateHTML: require('es-abstract/2020/CreateHTML'),
	CreateIterResultObject: require('es-abstract/2020/CreateIterResultObject'),
	CreateListFromArrayLike: require('es-abstract/2020/CreateListFromArrayLike'),
	CreateMethodProperty: require('es-abstract/2020/CreateMethodProperty'),
	DateFromTime: require('es-abstract/2020/DateFromTime'),
	DateString: require('es-abstract/2020/DateString'),
	Day: require('es-abstract/2020/Day'),
	DayFromYear: require('es-abstract/2020/DayFromYear'),
	DaysInYear: require('es-abstract/2020/DaysInYear'),
	DayWithinYear: require('es-abstract/2020/DayWithinYear'),
	DefinePropertyOrThrow: require('es-abstract/2020/DefinePropertyOrThrow'),
	DeletePropertyOrThrow: require('es-abstract/2020/DeletePropertyOrThrow'),
	EnumerableOwnPropertyNames: require('es-abstract/2020/EnumerableOwnPropertyNames'),
	FlattenIntoArray: require('es-abstract/2020/FlattenIntoArray'),
	floor: require('es-abstract/2020/floor'),
	FromPropertyDescriptor: require('es-abstract/2020/FromPropertyDescriptor'),
	Get: require('es-abstract/2020/Get'),
	GetIterator: require('es-abstract/2020/GetIterator'),
	GetMethod: require('es-abstract/2020/GetMethod'),
	GetOwnPropertyKeys: require('es-abstract/2020/GetOwnPropertyKeys'),
	GetPrototypeFromConstructor: require('es-abstract/2020/GetPrototypeFromConstructor'),
	GetSubstitution: require('es-abstract/2020/GetSubstitution'),
	GetV: require('es-abstract/2020/GetV'),
	HasOwnProperty: require('es-abstract/2020/HasOwnProperty'),
	HasProperty: require('es-abstract/2020/HasProperty'),
	HourFromTime: require('es-abstract/2020/HourFromTime'),
	InLeapYear: require('es-abstract/2020/InLeapYear'),
	InstanceofOperator: require('es-abstract/2020/InstanceofOperator'),
	Invoke: require('es-abstract/2020/Invoke'),
	IsAccessorDescriptor: require('es-abstract/2020/IsAccessorDescriptor'),
	IsArray: require('es-abstract/2020/IsArray'),
	IsBigIntElementType: require('es-abstract/2020/IsBigIntElementType'),
	IsCallable: require('es-abstract/2020/IsCallable'),
	IsConcatSpreadable: require('es-abstract/2020/IsConcatSpreadable'),
	IsConstructor: require('es-abstract/2020/IsConstructor'),
	IsDataDescriptor: require('es-abstract/2020/IsDataDescriptor'),
	IsExtensible: require('es-abstract/2020/IsExtensible'),
	IsGenericDescriptor: require('es-abstract/2020/IsGenericDescriptor'),
	IsInteger: require('es-abstract/2020/IsInteger'),
	IsNonNegativeInteger: require('es-abstract/2020/IsNonNegativeInteger'),
	IsNoTearConfiguration: require('es-abstract/2020/IsNoTearConfiguration'),
	IsPromise: require('es-abstract/2020/IsPromise'),
	IsPropertyKey: require('es-abstract/2020/IsPropertyKey'),
	IsRegExp: require('es-abstract/2020/IsRegExp'),
	IsStringPrefix: require('es-abstract/2020/IsStringPrefix'),
	IsUnclampedIntegerElementType: require('es-abstract/2020/IsUnclampedIntegerElementType'),
	IsUnsignedElementType: require('es-abstract/2020/IsUnsignedElementType'),
	IterableToList: require('es-abstract/2020/IterableToList'),
	IteratorClose: require('es-abstract/2020/IteratorClose'),
	IteratorComplete: require('es-abstract/2020/IteratorComplete'),
	IteratorNext: require('es-abstract/2020/IteratorNext'),
	IteratorStep: require('es-abstract/2020/IteratorStep'),
	IteratorValue: require('es-abstract/2020/IteratorValue'),
	LengthOfArrayLike: require('es-abstract/2020/LengthOfArrayLike'),
	MakeDate: require('es-abstract/2020/MakeDate'),
	MakeDay: require('es-abstract/2020/MakeDay'),
	MakeTime: require('es-abstract/2020/MakeTime'),
	MinFromTime: require('es-abstract/2020/MinFromTime'),
	modulo: require('es-abstract/2020/modulo'),
	MonthFromTime: require('es-abstract/2020/MonthFromTime'),
	msFromTime: require('es-abstract/2020/msFromTime'),
	NumberBitwiseOp: require('es-abstract/2020/NumberBitwiseOp'),
	OrdinaryCreateFromConstructor: require('es-abstract/2020/OrdinaryCreateFromConstructor'),
	OrdinaryDefineOwnProperty: require('es-abstract/2020/OrdinaryDefineOwnProperty'),
	OrdinaryGetOwnProperty: require('es-abstract/2020/OrdinaryGetOwnProperty'),
	OrdinaryGetPrototypeOf: require('es-abstract/2020/OrdinaryGetPrototypeOf'),
	OrdinaryHasInstance: require('es-abstract/2020/OrdinaryHasInstance'),
	OrdinaryHasProperty: require('es-abstract/2020/OrdinaryHasProperty'),
	OrdinaryObjectCreate: require('es-abstract/2020/OrdinaryObjectCreate'),
	OrdinarySetPrototypeOf: require('es-abstract/2020/OrdinarySetPrototypeOf'),
	PromiseResolve: require('es-abstract/2020/PromiseResolve'),
	QuoteJSONString: require('es-abstract/2020/QuoteJSONString'),
	RegExpExec: require('es-abstract/2020/RegExpExec'),
	RequireObjectCoercible: require('es-abstract/2020/RequireObjectCoercible'),
	SameValue: require('es-abstract/2020/SameValue'),
	SameValueNonNumeric: require('es-abstract/2020/SameValueNonNumeric'),
	SameValueZero: require('es-abstract/2020/SameValueZero'),
	SecFromTime: require('es-abstract/2020/SecFromTime'),
	Set: require('es-abstract/2020/Set'),
	SetFunctionLength: require('es-abstract/2020/SetFunctionLength'),
	SetFunctionName: require('es-abstract/2020/SetFunctionName'),
	SetIntegrityLevel: require('es-abstract/2020/SetIntegrityLevel'),
	SpeciesConstructor: require('es-abstract/2020/SpeciesConstructor'),
	StringGetOwnProperty: require('es-abstract/2020/StringGetOwnProperty'),
	StringPad: require('es-abstract/2020/StringPad'),
	SymbolDescriptiveString: require('es-abstract/2020/SymbolDescriptiveString'),
	TestIntegrityLevel: require('es-abstract/2020/TestIntegrityLevel'),
	thisBigIntValue: require('es-abstract/2020/thisBigIntValue'),
	thisBooleanValue: require('es-abstract/2020/thisBooleanValue'),
	thisNumberValue: require('es-abstract/2020/thisNumberValue'),
	thisStringValue: require('es-abstract/2020/thisStringValue'),
	thisSymbolValue: require('es-abstract/2020/thisSymbolValue'),
	thisTimeValue: require('es-abstract/2020/thisTimeValue'),
	TimeClip: require('es-abstract/2020/TimeClip'),
	TimeFromYear: require('es-abstract/2020/TimeFromYear'),
	TimeString: require('es-abstract/2020/TimeString'),
	TimeWithinDay: require('es-abstract/2020/TimeWithinDay'),
	ToBoolean: require('es-abstract/2020/ToBoolean'),
	ToDateString: require('es-abstract/2020/ToDateString'),
	ToIndex: require('es-abstract/2020/ToIndex'),
	ToInt16: require('es-abstract/2020/ToInt16'),
	ToInt32: require('es-abstract/2020/ToInt32'),
	ToInt8: require('es-abstract/2020/ToInt8'),
	ToInteger: require('es-abstract/2020/ToInteger'),
	ToLength: require('es-abstract/2020/ToLength'),
	ToNumber: require('es-abstract/2020/ToNumber'),
	ToNumeric: require('es-abstract/2020/ToNumeric'),
	ToObject: require('es-abstract/2020/ToObject'),
	ToPrimitive: require('es-abstract/2020/ToPrimitive'),
	ToPropertyDescriptor: require('es-abstract/2020/ToPropertyDescriptor'),
	ToPropertyKey: require('es-abstract/2020/ToPropertyKey'),
	ToString: require('es-abstract/2020/ToString'),
	ToUint16: require('es-abstract/2020/ToUint16'),
	ToUint32: require('es-abstract/2020/ToUint32'),
	ToUint8: require('es-abstract/2020/ToUint8'),
	ToUint8Clamp: require('es-abstract/2020/ToUint8Clamp'),
	TrimString: require('es-abstract/2020/TrimString'),
	Type: require('es-abstract/2020/Type'),
	UnicodeEscape: require('es-abstract/2020/UnicodeEscape'),
	UTF16DecodeString: require('es-abstract/2020/UTF16DecodeString'),
	UTF16DecodeSurrogatePair: require('es-abstract/2020/UTF16DecodeSurrogatePair'),
	UTF16Encoding: require('es-abstract/2020/UTF16Encoding'),
	ValidateAndApplyPropertyDescriptor: require('es-abstract/2020/ValidateAndApplyPropertyDescriptor'),
	WeekDay: require('es-abstract/2020/WeekDay'),
	YearFromTime: require('es-abstract/2020/YearFromTime')
};

module.exports = ES2020;
